tools/asset-sync/assets.ps1

```diff
@@ -0,0 +1,521 @@
+
+.PARAMETER Config
+A PSCustomObject that contains an auto-parsed assets.json content.
+
+#>
+Function Resolve-CheckoutPaths {
+    param(
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject] $Config
+    )
+    $assetsJsonPath = $Config.AssetsJsonRelativeLocation
+
+    $assetsJsonFolder = Split-Path $Config.AssetsJsonRelativeLocation
+
+    if(!$assetsJsonFolder -or $assetsJsonFolder -in @(".", "./", ".\"))
+    {
+        return $Config.AssetsRepoPrefixPath.Replace("`\", "/")
+    }
+    else {
+        $result = (Join-Path $Config.AssetsRepoPrefixPath $assetsJsonFolder).Replace("`\", "/")
+        return $result
+    }
+}
+
+<#
+.SYNOPSIS
+Given a configuration, determine the _current_ target path.
+
+.DESCRIPTION 
+Determines the presence of a branch on the git repo. If the relevant auto/<service> branch does not exist, we should use main.
+#>
+Function Resolve-TargetBranch {
+    param(
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject] $Config
+    )
+    $assetRepo = Resolve-AssetRepo-Location -Config $Config
+    $branch = "main"
+    try {
+        Push-Location $assetRepo
+        
+        $latestCommit = git rev-parse "origin/$($Config.AssetsRepoBranch)"
+
+        if($LASTEXITCODE -eq 0) {
+            $branch = $Config.AssetsRepoBranch
+        }
+    }
+    finally {
+        Pop-Location
+    }
+
+    return $branch
+}
+
+<#
+.SYNOPSIS
+Initializes a recordings repo based on an assets.json file. 
+
+.DESCRIPTION
+This Function will NOT re-initialize a repo if it discovers the repo already ready to go.
+
+.PARAMETER Config
+A PSCustomObject that contains an auto-parsed assets.json content from Resolve-AssetsJson.
+
+.PARAMETER ForceReinitialize
+Should this assets repo be renewed regardless of current status?
+#>
+Function Initialize-AssetsRepo {
+    param(
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject] $Config,
+        [Parameter(Mandatory=$false)]
+        [boolean] $ForceReinitialize = $false
+    )
+    $assetRepo = Resolve-AssetRepo-Location -Config $Config
+    $initialized = Is-AssetsRepo-Initialized -Config $Config
+    $workCompleted = $false
+
+    if ($ForceReinitialize)
+    {
+        Remove-Item -Force -R "$assetRepo/*"
+        $initialized = $false
+    }
+
+    if (!$initialized){
+        try {
+            Push-Location $assetRepo
+        
+            Write-Host "git clone --no-checkout --filter=tree:0 https://github.com/$($Config.AssetsRepo) ."
+            git clone --no-checkout --filter=tree:0 https://github.com/$($Config.AssetsRepo) .
+
+            $targetPath = Resolve-CheckoutPaths -Config $Config
+            $targetBranch = Resolve-TargetBranch -Config $Config
+
+            Write-Host "git sparse-checkout init"
+            git sparse-checkout init
+
+            Write-Host "git sparse-checkout set '/*' '!/*/' $targetPath"
+            git sparse-checkout set '/*' '!/*/' $targetPath
```

Comment:
I think you might be able to skip the first two globs.

Reply:
👍